dev/check-codegen-python.py

```diff
@@ -75,7 +75,9 @@ def check_connect_protos():
         else:
             fail(
                 "Generated files for pyspark-connect are out of sync! "
-                "Please run ./connector/connect/dev/generate_protos.sh"
+                "If you have touched files under connector/connect/src/main/protobuf, "
+                "please run ./connector/connect/dev/generate_protos.sh. "
+                "If you haven't touched any file above, please rebase your PR against main branch."
```

Comment:
How does rebase help? I think ci/cd merges with master.

Reply:
This script generates files from the proto itself, and proto might be updated in the main branch, so that the generated files in the sky now is different from the files in this local branch thus the check fails.

Reply:
Is this for CI/CD or for local tests? Rebasing helps local branch, yes, not for CI/CD.

Reply:
This is to help pass CI/CD. CI/CD sync to latest main branch if I understand it right: 

<img width="454" alt="Screen Shot 2022-10-21 at 1 55 14 PM" src="https://user-images.githubusercontent.com/1938382/197287176-e92cf3f5-7dc8-4273-bc8d-e3e0272327b3.png">


If don't consider CI/CD, the local development has no issue.